src/modules/csm/csm.py

```diff
@@ -224,13 +220,19 @@ def collect_data(self, blockstamp: BlockStamp) -> bool:
         return self.state.is_fulfilled
 
     def calculate_distribution(
-        self, blockstamp: ReferenceBlockStamp
-    ) -> tuple[Shares, defaultdict[NodeOperatorId, Shares], list[FramePerfLog]]:
+        self, blockstamp: ReferenceBlockStamp, last_report: "LastReport"
+    ) -> tuple[
+        Shares,
+        defaultdict[NodeOperatorId, Shares],
+        dict[StrikesValidator, StrikesList],
+        list[FramePerfLog],
+    ]:
         """Computes distribution of fee shares at the given timestamp"""
         operators_to_validators = self.module_validators_by_node_operators(blockstamp)
 
-        total_distributed = Shares(0)
         total_rewards = defaultdict[NodeOperatorId, Shares](Shares)
+        total_distributed = Shares(0)
+        strikes = last_report.strikes
```

Comment:
Does it mutate `last_report.strikes` as well? Can be a bit misleading during debug process. Let's copy it